packages/framework/src/Component/Elasticsearch/IndexDefinition.php

```diff
@@ -0,0 +1,132 @@
+     * @var \Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractIndex
+     */
+    protected $index;
+
+    /**
+     * @var string
+     */
+    protected $definitionsDirectory;
+
+    /**
+     * @var string
+     */
+    protected $indexPrefix;
+
+    /**
+     * @var int
+     */
+    protected $domainId;
+
+    /**
+     * DocumentDefinition constructor.
+     *
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractIndex $index
+     * @param string $definitionsDirectory
+     * @param string $indexPrefix
+     * @param int $domainId
+     */
+    public function __construct(AbstractIndex $index, string $definitionsDirectory, string $indexPrefix, int $domainId)
+    {
+        $this->index = $index;
+        $this->definitionsDirectory = $definitionsDirectory;
+        $this->indexPrefix = $indexPrefix;
+        $this->domainId = $domainId;
+    }
+
+    /**
+     * @return array
+     */
+    public function getDefinition(): array
+    {
+        $decodedDefinition = json_decode($this->getDefinitionFileContent(), true);
+        if ($decodedDefinition === null) {
+            throw ElasticsearchIndexException::invalidJsonInDefinitionFile(
+                $this->getIndex()->getName(),
+                $this->getDefinitionFilename()
+            );
+        }
+
+        return $decodedDefinition;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getDefinitionFilename(): string
+    {
+        return $this->getDefinitionsDirectory() . $this->getIndex()->getName() . '/' . $this->getDomainId() . '.json';
+    }
+
+    /**
+     * @return string
+     */
+    protected function getDefinitionFileContent(): string
+    {
+        $definitionFilename = $this->getDefinitionFilename();
+        if (!is_readable($definitionFilename)) {
+            throw ElasticsearchIndexException::cantReadDefinitionFile($definitionFilename);
+        }
+
+        return file_get_contents($definitionFilename);
+    }
+
+    /**
+     * @return string
+     */
+    public function getDocumentDefinitionVersion(): string
+    {
+        return md5(serialize($this->getDefinition()));
+    }
+
+    /**
+     * @return string
+     */
+    public function getVersionedIndexName(): string
+    {
+        return sprintf('%s_%s', $this->getIndexAlias(), $this->getDocumentDefinitionVersion());
+    }
+
+    /**
+     * @return string
+     */
+    public function getIndexAlias(): string
+    {
+        if ($this->indexPrefix === '') {
+            return sprintf('%s_%s', $this->getIndex()->getName(), $this->getDomainId());
+        }
+        return sprintf('%s_%s_%s', $this->indexPrefix, $this->getIndex()->getName(), $this->getDomainId());
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractIndex
+     */
+    public function getIndex(): AbstractIndex
+    {
+        return $this->index;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDefinitionsDirectory(): string
```

Comment:
Is this getter even necessary as it's only used in getDefinitionFilename?

Reply:
I personally do not like accessing properties directly. Even it seems unnecessary I would like keep it here. It may help us one day when we will want to change where documents are stored (for any reason) without affecting the property.

Reply:
After discussion we decided to access property directly inside its class and remove the getter.